src/Tes.Runner/Events/BlobStorageEventSink.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using Azure.Storage.Blobs;
+using Microsoft.Extensions.Logging;
+using Tes.Runner.Transfer;
+using static System.DateTime;
+
+namespace Tes.Runner.Events
+{
+    public class BlobStorageEventSink : EventSink
+    {
+        const string EventTimeStampFormat = "HH-mm-ss.fff";
+        const string Iso8601DateFormat = "yyyy-MM-ddTHH:mm:ss.fffZ";
+        private const string ApiVersion = "2023-05-03";
+        private readonly Uri storageUrl;
+        private readonly ILogger logger = PipelineLoggerFactory.Create<BlobStorageEventSink>();
+        private readonly BlobBlockApiHttpUtils blobBlockApiHttpUtils = new BlobBlockApiHttpUtils();
+
+
+        public BlobStorageEventSink(Uri storageUrl)
+        {
+            ArgumentNullException.ThrowIfNull(storageUrl);
+
+            this.storageUrl = storageUrl;
+        }
+
+        public override async Task HandleEventAsync(EventMessage eventMessage)
+        {
+            try
+            {
+                var content = JsonSerializer.Serialize(eventMessage);
+
+                await blobBlockApiHttpUtils.ExecuteHttpRequestAsync(() =>
+                    BlobBlockApiHttpUtils.CreatePutBlobRequestAsync(ToEventUrl(storageUrl, eventMessage), content, ApiVersion, ToTags(eventMessage)));
+            }
+            catch (Exception e)
+            {
+                //failure to publish event to blob storage should not fail the execution of the node task
+                logger.LogError(e, $"Failed to publish event {eventMessage.Id} to blob storage");
+            }
+        }
+
+        private string ToEventUrl(Uri uri, EventMessage message)
+        {
+            var blobBuilder = new BlobUriBuilder(uri);
+
+            var blobName = ToBlobName(message);
+
+            if (!string.IsNullOrWhiteSpace(blobBuilder.BlobName))
+            {
+                blobName = $"{blobBuilder.BlobName.TrimEnd('/')}/{blobName}";
```

Comment:
blobName.TrimStart('/') to be safe too?

Reply:
No need, `ToBlobName` always returns a value that starts with `events/...`